staging/src/k8s.io/endpointslice/util/controller_utils.go

```diff
@@ -297,8 +296,58 @@ func stringPtrChanged(ptr1, ptr2 *string) bool {
 	return false
 }
 
-// EndpointSubsetsEqualIgnoreResourceVersion returns true if EndpointSubsets
-// have equal attributes but excludes ResourceVersion of Pod.
-func EndpointSubsetsEqualIgnoreResourceVersion(subsets1, subsets2 []v1.EndpointSubset) bool {
-	return semanticIgnoreResourceVersion.DeepEqual(subsets1, subsets2)
+// DeepHashObject writes specified object to hash using the spew library
+// which follows pointers and prints actual values of the nested objects
+// ensuring the hash does not change when a pointer changes.
+// copied from k8s.io/kubernetes/pkg/util/hash
+func deepHashObject(hasher hash.Hash, objectToWrite interface{}) {
+	hasher.Reset()
+	printer := spew.ConfigState{
+		Indent:         " ",
+		SortKeys:       true,
+		DisableMethods: true,
+		SpewKeys:       true,
+	}
+	printer.Fprintf(hasher, "%#v", objectToWrite)
+}
+
+// IsPodReady returns true if Pods Ready condition is true
+// copied from k8s.io/kubernetes/pkg/api/v1/pod
```

Comment:
@thockin @BenTheElder do you know some magic to keep these things in sync?
some linter maybe?

Reply:
no, I don't have a good answer.  Most of them are API based, and we won't intentionally break that, so it's another case "a little copy is better than a little dep", probably.